src/engine/BMPlayer.php

```diff
@@ -0,0 +1,264 @@
+<?php
+
+/**
+ * BMPlayer: game-relevant attributes of a player
+ *
+ * @author james
+ */
+
+/**
+ * This class contains all the player-related attributes of a game
+ *
+ * @property      int      $playerId               Player ID number in the database
+ * @property      int      $position               Position index of this player in this game
+ * @property      BMButton $button                 Button used by this player in this game
+ * @property      array    $activeDieArray         Array of active dice owned by this player
+ * @property      array    $capturedDieArray       Array of dice captured by this player
+ * @property      array    $outOfPlayDieArray      Array of dice out of play
+ * @property      bool     $waitingOnAction        Does this player need to perform an action?
+ * @property      bool     $isPrevRoundWinner      Has this player just won the previous round?
+ * @property      BMGame   $ownerObject            BMGame that owns this BMPlayer object
+ */
+class BMPlayer {
+    // properties -- all accessible, but written as protected to enable the use of
+    //               getters and setters
+
+    /**
+     * Player ID number in the database
+     *
+     * @var int
+     */
+    protected $playerId;
+
+    /**
+     * Position index of this player in this game
+     *
+     * @var int
+     */
+    protected $position;
+
+    /**
+     * Button used by the player in this game
+     *
+     * @var BMButton
+     */
+    protected $button;
+
+    /**
+     * Array of active dice owned by this player
+     *
+     * This is deliberately public, since PHP doesn't allow proper access to
+     * the contents of encapsulated arrays.
+     *
+     * @var array
+     */
+    public $activeDieArray;
+
+    /**
+     * Array of dice captured by this player
+     *
+     * This is deliberately public, since PHP doesn't allow proper access to
+     * the contents of encapsulated arrays.
+     *
+     * @var array
+     */
+    public $capturedDieArray;
+
+    /**
+     * Array of dice out of play, but nominally belonging to this player
+     *
+     * This is deliberately public, since PHP doesn't allow proper access to
+     * the contents of encapsulated arrays.
+     *
+     * @var array
+     */
+    public $outOfPlayDieArray;
+
+    /**
+     * Boolean specifying if this player needs to perform an action at this time
+     *
+     * @var bool
+     */
+    protected $waitingOnAction;
+
+    /**
+     * Boolean specifying if this player won the previous round
+     *
+     * @var bool
+     */
+    protected $isPrevRoundWinner;
+
+    /**
+     * BMGame that owns this BMPlayer object
+     *
+     * @var BMGame
+     */
+    protected $ownerObject;
+
+
+    // setter methods
+    /**
+     * Set ID
+     *
+     * @param int|null $value
+     */
+    protected function set__playerId($value) {
+        if (!is_null($value)) {
```

Comment:
I wonder if I'll actually break something if I enforce this to be an int, considering PHP's weak typing.  :)  I'll take a look.